### ISSUE TITLE
FillIPolicy silently discards unexpected items

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#457]
Originally reported of Google Code with ID 401.

```
objects.FillIPolicy() silently discards any unexpected item in the policy to fill.

This is problem when upgrading (part of the old configuration can be dropped), and when performing validation, as bad data could be present in the configuration but hidden from validating functions that call FillIPolicy().

```
Originally added on 2013-03-26 22:14:31 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This has been fixed as a side effect of commit 318b19fe, but unit tests are still missing:

commit 318b19fe8912667112b93ca2adb045829ca690d2
Author: Bernardo Dal Seno <bdalseno@google.com>
Date:   Mon Apr 8 20:36:34 2013 +0200

    Limit specs in instance policies are always complete
    
    Specs used to specify limits in instance policies are always complete, even
    at group level (only the whole limit set can be overridden). This is in
    preparation for introducing multiple limits.

```
Originally added on 2013-05-03 09:57:19 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit f73f35673c48d84d88e92f178dd4394134a7f623
Author: Bernardo Dal Seno <bdalseno@google.com>
Date:   Fri May 3 11:51:06 2013 +0200

    Unit test for FillIPolicy keeping unknown keys
    
    This makes sure that issue 401 doesn't come back.
    
    Signed-off-by: Bernardo Dal Seno <bdalseno@google.com>
    Reviewed-by: Helga Velroyen <helgav@google.com>

```
Originally added on 2013-05-03 12:39:55 +0000 UTC.

Mentioned Issue with GoogleCode ID 401 has been migrated to GitHub with ID #457.
Changed State: Fixed
Added to Milestone: Release2.8
